packages/property-view/src/browser/resource-property-view/resource-property-data-service.spec.ts

```diff
@@ -0,0 +1,127 @@
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { enableJSDOM } from '@theia/core/lib/browser/test/jsdom';
+
+let disableJSDOM = enableJSDOM();
+
+import { expect } from 'chai';
+import { FrontendApplicationConfigProvider } from '@theia/core/lib/browser/frontend-application-config-provider';
+import { ApplicationProps } from '@theia/application-package/lib/application-props';
+FrontendApplicationConfigProvider.set({
+    ...ApplicationProps.DEFAULT.frontend.config
+});
+
+import { Container } from '@theia/core/shared/inversify';
+import { ResourcePropertyDataService } from './resource-property-data-service';
+import { FileService } from '@theia/filesystem/lib/browser/file-service';
+import URI from '@theia/core/lib/common/uri';
+import { PropertyDataService } from '../property-data-service';
+import { FileSelection } from '@theia/filesystem/lib/browser/file-selection';
+import { Navigatable } from '@theia/core/lib/browser/navigatable';
+import { FileStat } from '@theia/filesystem/lib/common/files';
+
+disableJSDOM();
+
+let resourcePropertyDataService: ResourcePropertyDataService;
+
+const mockFileStat: FileStat = {
+    isFile: false,
+    isDirectory: true,
+    isSymbolicLink: false,
+    resource: new URI('resource'),
+    name: 'name'
+};
+
+describe('resource-property-data-service', () => {
+
+    before(() => {
+        disableJSDOM = enableJSDOM();
+        const container = new Container();
+        container.bind(ResourcePropertyDataService).toSelf().inSingletonScope();
+        container.bind(FileService).toConstantValue({
+            async resolve(uri: URI): Promise<FileStat> {
+                return mockFileStat;
+            }
+        } as FileService);
+        container.bind(PropertyDataService).to(ResourcePropertyDataService).inSingletonScope();
+        resourcePropertyDataService = container.get(ResourcePropertyDataService);
+    });
+
+    after(() => {
+        disableJSDOM();
+    });
+
+    const fileSelection: FileSelection = {
+        fileStat: mockFileStat
+    };
+
+    const navigatableSelection: Navigatable = {
+        getResourceUri(): URI | undefined {
+            return new URI('resource-uri');
+        },
+        createMoveToUri(): URI | undefined {
+            return new URI('move-uri');
+        }
+    };
+
+    const arrayFileSelection: FileSelection[] = [fileSelection, { fileStat: { ...mockFileStat, resource: new URI('secondURI') } }];
```

Comment:
We can reformat the line to make it more readable, in addition `fileSelection` is only ever used once so it can be removed:

```suggestion
    const arrayFileSelection: FileSelection[] = [
        { filestat: mockFileStat },
        { fileStat: { ...mockFileStat, resource: new URI('secondURI') } }
    ];
```